static/js/input_pill.js

```diff
@@ -96,25 +98,57 @@ export function create(opts) {
 
             store.pills.push(payload);
 
-            const has_image = item.img_src !== undefined;
+            if (item.type === "user" && !people.is_person_active(item.user_id)) {
+                payload.$element = this.renderDeactivatedUserData(payload);
+            } else {
+                payload.$element = this.renderActiveUserData(payload);
+            }
+
+            store.$input.before(payload.$element);
+        },
+
+        renderActiveUserData(payload) {
+            const has_image = payload.item.img_src !== undefined;
 
             const opts = {
                 id: payload.id,
-                display_value: item.display_value,
+                display_value: payload.item.display_value,
                 has_image,
             };
 
             if (has_image) {
-                opts.img_src = item.img_src;
+                opts.img_src = payload.item.img_src;
             }
 
             if (typeof store.onPillCreate === "function") {
                 store.onPillCreate();
             }
 
             const pill_html = render_input_pill(opts);
-            payload.$element = $(pill_html);
-            store.$input.before(payload.$element);
+            return $(pill_html);
+        },
+
+        renderDeactivatedUserData(payload) {
+            const has_image = payload.item.img_src !== undefined;
+
+            const opts = {
+                id: payload.id,
+                display_value: payload.item.display_value + " (deactivated)",
+                has_image,
+            };
+
+            if (has_image) {
+                opts.img_src = payload.item.img_src;
+            }
+
+            const pill_html = render_input_pill(opts);
+            const $pill = $(pill_html);
+            $pill.addClass("deactivated-user-pill");
+            $pill.prop(
+                "title",
+                $t({defaultMessage: "Can't send messages to the deactivated user"}),
+            );
+            return $pill;
```

Comment:
This function seems to mostly duplicate the original; could we just have `is_active` be a parameter to the "render a user pill" function?

Reply:
I have created an option `deactivated_user_pill` that handles this properly. Yes I was also feeling that the code was almost repeated.